File: Chapter 1/Section 1.2/ride/ride.py
"""
ID: deadly51
TASK: ride
LANG: PYTHON3

"""
def prod(s):
    temp = 1
    for i in s:
        temp *= i
    return temp

def convert(s):
    return [ord(i)%32 for i in s]

def compare(s1, s2):
    return "GO\n" if prod(convert(s1)) %47 == prod(convert(s2)) %47 else "STAY\n"


fin = open("ride.in","r")
fout = open("ride.out", "w") 
str1 = fin.readline().strip()
str2 = fin.readline().strip()
fin.close()
fout.write(compare(str1, str2))
fout.close()

File: Chapter 1/Section 1.2/ride/ride.cpp

/*
ID: sagosag1
TASK: ride
LANG: C++
*/

#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main() {
	ofstream fout ("ride.out");
	ifstream fin ("ride.in");

	string s1,s2;
	fin >> s1 >> s2;

	int prod1 = 1, prod2 = 1;
	for (auto& c: s1) {
		prod1 *= c-'A'+1;
	}
	for (auto& c: s2) {
		prod2 *= c-'A'+1;
	}

	if (prod1 % 47 == prod2 % 47) {
		fout << "GO\n";
	} else {
		fout << "STAY\n";
	}

	return 0;
}

